src/test/java/com/rocketden/main/service/RoomServiceTests.java

```diff
@@ -345,13 +379,15 @@ public void updateRoomSettingsSuccess() {
         request.setInitiator(UserMapper.toDto(host));
         request.setDifficulty(ProblemDifficulty.EASY);
         request.setDuration(DURATION);
+        request.setSize(5);
         request.setNumProblems(3);
 
         RoomDto response = roomService.updateRoomSettings(room.getRoomId(), request);
 
         verify(socketService).sendSocketUpdate(eq(response));
         assertEquals(request.getDifficulty(), response.getDifficulty());
         assertEquals(request.getDuration(), response.getDuration());
+        assertEquals(request.getSize(), response.getSize());
         assertEquals(request.getNumProblems(), response.getNumProblems());
```

Comment:
Let me know if you've already done this (such as in the previous PR), but could you add 2 API tests (in the RoomTests.java class), one a failure to join a room when max limit is reached and another for when trying to set an invalid room size (such as less than the number of current players for example)? As well, if there are existing tests such as updateRoomSettingsSuccess, then maybe just add two quick lines to those for setting the max size and checking with an assert in the returned object that the field is set properly

Reply:
I added the tests to the RoomServiceTests.java file. Is that fine instead?

Reply:
We have tests in two places, both the unit tests and the integration tests. There is some redundancy here and we've talked about this before - if you think we should probably change habits and just use the service tests we can talk on the call. Thus far we've been doing both, but happy to chat more during the daily standup.